src/components/category/updateCategoryModal.js

```diff
@@ -0,0 +1,245 @@
+    });
+  };
+
+  onFormSubmit = (e) => {
+    const { name, description } = this.state;
+    e.preventDefault();
+    this.props.updateCategory(this.props.category._id, name, description);
+  };
+
+  handleDeleteModalShow = () => {
+    this.setState({
+      showDeleteModal: true,
+    });
+  };
+
+  handleDeleteModalClose = () => {
+    this.setState({
+      showDeleteModal: false,
+    });
+  };
+
+  onDeleteSubmit = () => {
+    this.props.deleteCategory(this.props.category._id);
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      nameError,
+      descriptionError,
+      formSubmissionError,
+      isFormInvalid,
+    } = this.state;
+    const {
+      name,
+      description,
+    } = this.props.category;
+    let newState = {};
+    if (prevProps.showModal != this.props.showModal) {
+      newState = {
+        ...newState,
+        name: name,
+        nameError: null,
+        description: description,
+        descriptionError: null,
+        isFormInvalid: true,
+      };
+      if (formSubmissionError != "") {
+        newState = {
+          ...newState,
+          formSubmissionError: "",
+        };
+      }
+    } else {
+      if (this.props.error != prevProps.error) {
+        newState = {
+          ...newState,
+          formSubmissionError: this.props.error,
+        };
+      }
+      if (isFormInvalid == true) {
+        if (nameError === null && descriptionError === null) {
+          newState = {
+            ...newState,
+            isFormInvalid: false,
+          };
+        }
+      } else if (nameError !== null || descriptionError !== null) {
+        newState = {
+          ...newState,
+          isFormInvalid: true,
+        };
+      }
+    }
+    if (Object.keys(newState).length != 0) {
+      this.setState(newState);
+    }
+    if (
+      !prevProps.isUpdateCompleted &&
+      prevProps.isUpdateCompleted != this.props.isUpdateCompleted
+    ) {
+      this.props.getCategories();
+      this.props.handleClose("updateCategory");
+    }
+    if (
+      !prevProps.isDeleteCompleted &&
+      prevProps.isDeleteCompleted != this.props.isDeleteCompleted
+    ) {
+      this.props.getCategories();
+      this.handleDeleteModalClose();
+      this.props.resetCurrentCategory();
+    }
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        <Modal
+          show={this.props.showModal}
+          onHide={() => {
+            this.props.handleClose("updateCategory");
+          }}
+          centered
+        >
+          <Modal.Body>
+            <Container>
+              <Row className="center-row">
+                <Col xs={12}>
+                  <h1 className="modal-heading">Edit Category</h1>
+                </Col>
+              </Row>
+              <Row>
+                <Col xs={12}>
+                  <div className="modal-form">
+                    {this.state.formSubmissionError && (
+                      <div className="alert alert-danger" role="alert">
```

Comment:
Any specific reasons for using this instead of using `<Alert/>`?

Reply:
Wasn't aware about that component. Will add fixes to next PR since I already have a branch based off this which will then accumulate merge conflicts

Reply:
Resolved in #29